src/redux/api/taskStepApi.tsx

```diff
@@ -12,15 +12,39 @@ export const taskStepApi = baseApi.injectEndpoints({
       }),
     }),
 
-    updateTaskStep: builder.mutation<Task,{id:string, changes: EditTaskStepRequest}>({
+    updateTaskStep: builder.mutation<
+      Task,
+      { id: string; changes: EditTaskStepRequest }
+    >({
       query: ({ id, changes }) => ({
         url: `/task-steps/${id}/edit/`,
         method: 'PUT',
         body: changes,
       }),
       invalidatesTags: ['Step'],
-    })
+    }),
+
+    deleteTaskStep: builder.mutation<void, { id: string }>({
+      query: ({ id }) => ({
+        url: `/task-steps/${id}/`,
+        method: 'DELETE',
+      }),
+      invalidatesTags: ['Step'],
```

Comment:
instead of Step, invalidate Task because the steps are fetch from the Task api not the Step. Since this invalidate tag purpose is for refetching the data with the current Tag. Additionally, if the response contain the task_id, you can add it like

invalidateTags: (result)=> [{type:'Task', id:result?.task_id ?? ''}]

This will refetch only the task with specified id, but make sure that in the get list there is provided tags to support this

Reply:
The idea was to refetch the steps after deleting, but i guess you have a point. Thx.